src/main/java/com/aws/greengrass/clientdevices/auth/api/ClientDevicesAuthServiceApi.java

```diff
@@ -49,8 +54,12 @@ public ClientDevicesAuthServiceApi(SessionManager sessionManager, DeviceAuthClie
     public boolean verifyClientDeviceIdentity(String certificatePem) {
         // Allow internal clients to verify their identities
         if (deviceAuthClient.isGreengrassComponent(certificatePem)) {
+            domainEvents.emit(new VerifyClientDeviceIdentityEvent(
```

Comment:
thinking out loud, would it be worth making this a generic `ApiOperationCompleteEvent`? that way we wouldn't need a `VerifyClientDeviceIdentityEvent.VerificationStatus` for each operation?  thoughts @jbutler @sameerzuberi

Reply:
I did consider doing that, the only reason I created a specific event and handler for the API is in case we decide we want to track other API specific things in the future. That way it'd be easy to just add a field in the enum for the event. But for right now since we're only focused on successes and failures it could work if we don't see ourselves making changes anytime soon

Reply:
yeah i agree having one for each would be more flexible---ignore my thought for now because we can always refactor later